account_product_fiscal_classification/models/product_template.py

```diff
@@ -74,16 +76,43 @@ def _update_vals_fiscal_classification(self, vals):
                     "taxes_id": [(6, 0, classification.sale_tax_ids.ids)],
                 }
             )
-        elif vals.get("supplier_taxes_id") or vals.get("taxes_id"):
-            raise ValidationError(
-                _(
-                    "You can not create or write products with"
-                    " 'Customer Taxes' or 'Supplier Taxes'\n."
-                    "Please, use instead the 'Fiscal Classification' field."
-                )
-            )
+        elif create_mode or {"supplier_taxes_id", "taxes_id"} & vals.keys():
+            self._find_or_create_classification(vals)
         return vals
 
     @api.constrains("categ_id", "fiscal_classification_id")
     def _check_rules_fiscal_classification(self):
         self.env["account.product.fiscal.rule"].check_product_templates_integrity(self)
+
+    def _find_or_create_classification(self, vals):
+        """Find the correct Fiscal classification,
+        depending of the taxes, or create a new one, if no one are found."""
+        # search for matching classication
+        purchase_tax_ids = vals.get("supplier_taxes_id", [])
+        sale_tax_ids = vals.get("taxes_id", [])
+        for elm in ("supplier_taxes_id", "taxes_id"):
+            if elm in vals:
+                del vals[elm]
+        domain = [("sale_tax_ids", "=", [])]
+        if sale_tax_ids:
+            domain = [("sale_tax_ids", "in", sale_tax_ids)]
+        domain.append(("purchase_tax_ids", "in", purchase_tax_ids or []))
```

Comment:
I don't understand this algorithm.

This looks like : 

domain = [("sale_tax_ids", "in", sale_tax_ids), ("purchase_tax_ids", "in", purchase_tax_ids)]

Don't you think ?